### PR TITLE
chore: Adds a11y badge, automated tests to numerous components

### PR DESCRIPTION
**Pull request by craigcarlyle**
## Short description

Adds the a11y badge and automated tests to:
- Box
- Button
- Columns
- Heading
- Hidden
- Inline
- PasswordField
- SelectField
- Stack
- Tabs
- Text
- TextField
- Tooltip

## PR Checklist

-   [x] Added tests for bugs / new features
-   [x] Updated docs (storybooks, readme)
-   [x] Executed `npm run validate` and made sure no errors / warnings were shown
-   [x] Updated all static build artifacts (`npm run build-all`)

## Versioning
This does not immediately require a new release.

### CODE REVIEW

File: src/new-components/box/box.test.tsx
Comment (gnapse) [inline src/new-components/box/box.test.tsx:264]: Is this a formality? For completeness? What's the value of testing that raw `div` renders without a11y violations?

File: src/new-components/button/button.test.tsx
Comment (gnapse) [inline src/new-components/button/button.test.tsx:367]: There's value in doing more tests for the button in other states: while disabled and while loading? These do affect some a11y related attributes. Maybe you can add those buttons to be rendered in this very test without adding more test cases.

I also think that icon-only buttons deserve their own test case, mirroring this test case, but rendering all buttons as icon-only buttons.

File: src/new-components/password-field/password-field.test.tsx
Comment (gnapse) [inline src/new-components/password-field/password-field.test.tsx:268]:
```suggestion
                <PasswordField label="New Password" />,
```

File: src/new-components/password-field/password-field.test.tsx
Comment (gnapse) [inline src/new-components/password-field/password-field.test.tsx:273]: I think this could be expanded to also check for a11y violations in password fields that have a `hint` (because this is supposed to add a `aria-describedby` attribute), and maybe test password fields that are disabled.

This feedback here applies to most of the other field components (`SelectField`, `TextField`). I won't repeat the same comment in those components, but whatever you decide to take from this suggestion here should be replicated in those other too.

File: src/new-components/select-field/select-field.test.tsx
Comment (gnapse) [inline src/new-components/select-field/select-field.test.tsx:201]:
```suggestion
                <SelectField label="Province" />,
```

File: src/new-components/text-field/text-field.test.tsx
Comment (gnapse) [inline src/new-components/text-field/text-field.test.tsx:190]: Test it without hint, to be consistent with other field tests, and with a hint as well.

File: stories/components/Dropdown.stories.tsx
Comment (gnapse) [inline stories/components/Dropdown.stories.tsx:10]: I would deprecate this one as well. I'm pretty certain that we're going to introduce a different component for this purpose.

File: stories/components/Time.stories.tsx
Comment (gnapse) [inline stories/components/Time.stories.tsx:15]: Does this one have jest-axe tests already?

File: stories/components/Tooltip.stories.tsx
Comment (gnapse) [inline stories/components/Tooltip.stories.tsx:14]: Does this one have jest-axe tests already?

**Review by gnapse — APPROVED** [review]
I'm approving, although I think that some of my comments need to be acted on in one way or another. But I'll leave it to you to decide, and no need to wait for a second review after that.

File: src/new-components/box/box.test.tsx
Comment (craigcarlyle) [reply src/new-components/box/box.test.tsx:264]: > Is this a formality? For completeness?

Correct.

File: stories/components/Time.stories.tsx
Comment (craigcarlyle) [reply stories/components/Time.stories.tsx:15]: Nope, added.

File: stories/components/Tooltip.stories.tsx
Comment (craigcarlyle) [reply stories/components/Tooltip.stories.tsx:14]: Nope, added.

**Pull request merged**
